app/models/user.rb

```diff
@@ -0,0 +1,19 @@
+require 'bcrypt'
+
+class User < ApplicationRecord
+  include BCrypt
+
+  has_many :posts
+  before_save :secure_password
+
+  validates :username, presence: true, uniqueness: { case_sensitive: false },
+                       length: { minimum: 4, maximum: 12 }
```

Comment:
- [OPTIONAL] You can use length: { in 4..12 } to tell your model to accept any input from length 4 to 12.